analyzers/tests/SonarAnalyzer.UnitTest/TestCases/UnusedStringBuilder.cs

```diff
@@ -0,0 +1,101 @@
+﻿using System.Collections.Generic;
+using System.Text;
+
+public class Program
+{
+    public StringBuilder MyMethod(StringBuilder builder, string myString) // Compliant
+    {
+        StringBuilder sb1 = GetStringBuilder(); // Compliant
+
+        var sb2 = new StringBuilder(); // Compliant
+        ExternalMethod(sb2);
```

Comment:
Needs a test for extension methods as well (probably an FP for now. Create an issue we can fix later).
```cs
sb.ToStringAndFree(); // Noncomliant FP

public static class StringBuilderExtensions
{
    string ToStringAndFree(this StringBuilder sb)
}
```

Fun Fact: Inspired by Roslyns PoolStringBuilder:
https://github.com/dotnet/roslyn/blob/315c2e149ba7889b0937d872274c33fcbfe9af5f/src/Dependencies/PooledObjects/PooledStringBuilder.cs#L56

Reply:
Nice! 👍 See https://github.com/SonarSource/sonar-dotnet/issues/6747